src/Aspire.Hosting.Azure.ServiceBus/AzureServiceBusExtensions.cs

```diff
@@ -30,23 +30,33 @@ public static IResourceBuilder<AzureServiceBusResource> AddAzureServiceBus(this
     {
         builder.AddAzureProvisioning();
 
-        var configureInfrastructure = static (AzureResourceInfrastructure infrastructure) =>
+        var configureInfrastructure = (AzureResourceInfrastructure infrastructure) =>
         {
             var skuParameter = new ProvisioningParameter("sku", typeof(string))
             {
                 Value = "Standard"
             };
             infrastructure.Add(skuParameter);
 
-            var serviceBusNamespace = new AzureProvisioning.ServiceBusNamespace(infrastructure.AspireResource.GetBicepIdentifier())
+            AzureProvisioning.ServiceBusNamespace? serviceBusNamespace;
+            if (infrastructure.AspireResource.TryGetExistingResource(out var existingAnnotation) &&
```

Comment:
Is there a way to resolve the execution context from the `AzureResourceInfrastructure` type. Is there a way to do this

Reply:
There may not be but this method has access to the builder in a closure right?